File: apple/Sources/CCDAEngine/Models/Common/CCDATemplateId.swift
import Foundation

/// C-CDA template identifier.
public struct CCDATemplateId: Hashable, Sendable {
    /// Template root OID.
    public let root: String
    /// Template version or extension value.
    public let extensionValue: String?

    /// Creates a template identifier.
    public init(root: String, extensionValue: String? = nil) {
        self.root = root
        self.extensionValue = extensionValue
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Common/CCDATemplateId.kt
package com.ccdakit.engine

/**
 * CDA template identifier used to declare document, section, or entry templates.
 *
 * @property root Template OID.
 * @property extensionValue Optional template version or extension.
 */
data class CCDATemplateId(
    val root: String,
    val extensionValue: String? = null,
)
